packages/react-sdk/src/components/image-dev.stories.tsx

```diff
@@ -0,0 +1,200 @@
+// Story for image development, see https://github.com/webstudio-is/webstudio-designer/issues/387
+
+import React, { type ComponentProps, type HTMLAttributes } from "react";
+import type { ComponentMeta, ComponentStory } from "@storybook/react";
+import { Image as ImagePrimitive } from "./image";
+import { type ImageLoader } from "../component-utils/image";
+import argTypes from "./image.props.json";
+
+// to not allow include local assets everywhere, just enable it for this file
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import localLogoImage from "../../storybook-assets/logo.webp";
+
+export default {
+  title: "Components/ImageDev",
+  component: ImagePrimitive,
+  argTypes,
+} as ComponentMeta<typeof ImagePrimitive>;
+
+type ImageProps = ComponentProps<typeof ImagePrimitive>;
+
+/**
+ * In case you need to test img with real cloudflare trasforms
+ * set  USE_CLOUDFLARE_IMAGE_TRANSFORM = true
+ **/
+const USE_CLOUDFLARE_IMAGE_TRANSFORM = false;
+
+// For cloudflare image transform testing, logo should be the most consistent image on the site
+const REMOTE_SELF_DOMAIN_IMAGE = "https://webstudio.is/logo.webp";
+
+const imageSrc = USE_CLOUDFLARE_IMAGE_TRANSFORM
+  ? REMOTE_SELF_DOMAIN_IMAGE
+  : localLogoImage;
+
+// In case of REMOTE_DEBUG
+const cloudflareImageLoader: ImageLoader = ({ width, src, quality }) => {
+  // No image transformation in development
+  const resizeOrigin = "https://webstudio.is";
+
+  const pathParams = [
+    `width=${width}`,
+    quality != null && `quality=${quality}`,
+    "format=auto",
+  ].filter(Boolean);
+
+  const pathname = `/cdn-cgi/image/${pathParams.join(",")}/${src}`;
+
+  const url = new URL(pathname, resizeOrigin);
+  return url.href;
+};
+
+const localImageLoader: ImageLoader = ({ width, src, quality }) => {
+  // Just emulate like we really resize the image
+  const params = new URLSearchParams();
+  params.set("width", `${width}`);
+  params.set("quality", `${quality}`);
+  return `${src}?${params.toString()}`;
+};
+
+const imageLoader = USE_CLOUDFLARE_IMAGE_TRANSFORM
+  ? cloudflareImageLoader
+  : localImageLoader;
+
+const ImageTemplate: ComponentStory<
+  React.ForwardRefExoticComponent<
+    ImageProps & { style: HTMLAttributes<"img">["style"] }
+  >
+> = (args) => {
+  const style = { maxWidth: "100%", ...args.style };
+
+  return (
+    <ImagePrimitive
+      {...args}
+      optimize={true}
+      loader={imageLoader}
+      style={style}
+    />
+  );
+};
+
+export const FixedWidthImage = ImageTemplate.bind({});
```

Comment:
btw we probably don't need to use template and bind thing, see for example packages/design-system/src/components/select.stories.tsx

Reply:
Found serious issue with inline styles during the rewrite, changed default image display type on `block`.